hack/verify-contextual-logging.sh

```diff
@@ -0,0 +1,44 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 The KCP Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o nounset
+set -o pipefail
+# Don't set errexit because logcheck's messages are in stderr
+
+SHOW=${1:-}
+
+SCRIPT_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
+# All subdirectories under ./pkg containing *.go files, _except_ in pkg/apis.
+# logcheck doesn't like that our main module doesn't reference pkg/apis.
```

Comment:
Can we run it inside of `pkg/apis` as well?

Reply:
We can if we use `logcheck -check-contextual ./...` instead of the find command - I'll switch over to that because it's a lot cleaner and gives us more coverage.

Reply:
You actually have to run it twice because pkg/apis is a different module, so you'll need to `cd` into it first and then run `./...` there too

Reply:
Ahh, ok, that makes sense for why I was seeing my errors with the find command too. Will add the 2nd call and combine the results.